src/Service/DefaultBlockInstaller.php

```diff
@@ -144,6 +144,13 @@ public function install(string $module): void {
     }
   }
 
+  /**
+   * Replace characters that aren't allowed in config IDs.
+   */
+  protected function sanitiseID(string $id): string {
+    return preg_replace('/[^a-z0-9\._]/', '_', strtolower($id));
```

Comment:
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Block%21BlockPluginTrait.php/function/BlockPluginTrait%3A%3AgetMachineNameSuggestion/11.x

Also removes not alphanumerical chars from the end, and uses mb_strtolower. I'm assuming we can't just use the trait here, but, probably should be doing the same thing?

Reply:
Good find! I've made it do the same thing, and added a comment referencing that method.

Reply:
Nice one.